### ISSUE TITLE
[feature request] support calculate_vm_cloud_properties

### ISSUE THREAD

**Issue by cppforlife** [cloudfoundry-incubator/bosh-google-cpi-release#253]
bosh director now supports more abstract vm size specification. users can specify on an instance group:

```
vm_resources:
  cpu: 2
  ram: 4096
  ephemeral_disk_size: 10000
```

instead of specifying separate `vm_type`. to support this in the cpi, one has to implement `calculate_vm_cloud_properties` cpi method which is called to translate generic definition to something create_vm can understand in cloud properties.

see http://bosh.io/docs/cpi-api-v1#calculate-vm-cloud-properties for more details.

example impl: https://github.com/cloudfoundry-incubator/bosh-vsphere-cpi-release/blob/master/src/vsphere_cpi/lib/cloud/vsphere/cloud.rb#L325 in vsphere which support one to one mapping.

we would like to get this added to google cpi hopefully sooner rather than later so that cf-deployment and other common deployments can update their manifests.

**Comment by johnsonj** [comment]
looks good! will go great with our [custom machine types](https://cloud.google.com/custom-machine-types/)

**Comment by johnsonj** [comment]
released as v27.0.0
